src/table/Root.tsx

```diff
@@ -1,31 +1,36 @@
 import React from 'react';
+import ReactDom from 'react-dom/client';
 import { StyleSheetManager } from 'styled-components';
 import { ThemeProvider } from '@mui/material/styles';
 import rtlPluginSc from 'stylis-plugin-rtl-sc';
+
 import TableWrapper from './components/TableWrapper';
 import { TableContextProvider } from './context';
 import muiSetup from './mui-setup';
+import { RenderProps, RootProps } from '../types';
 
-export function render(reactRoot, props) {
+export function render(reactRoot: ReactDom.Root | HTMLElement, props: RenderProps) {
   const { direction, selectionsAPI } = props;
   const muiTheme = muiSetup(direction);
+  const root = reactRoot as ReactDom.Root;
+  const rootProps = props as RootProps;
```

Comment:
Is it the same thing here, that you have to take both implementations into account?

Reply:
Right

Reply:
In practice, HTMLElement never reaches reactRoot here. But in typescript, we need to take care it.